pgwire: handle decoding Geometry/Geography in binary

otan opened the pull request:
Resolves #81066
Resolves #93352

Release note (bug fix): Previously, CockroachDB would error when receiving Geometry/Geography using binary parameters. This is now resolved.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/93563)
<!-- Reviewable:end -->

rafiss reviewed (approved):
thanks for adding it! would you be up for adding tests in pkg/cmd/generate-binary/main.go and `TestEncodings` in pkg/sql/pgwire/encoding_test.go

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/93563#-:-NJEcdVS0rpYtvnPkCpz:b-igdgg4)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
pgwire: handle decoding Geometry/Geography in binary

Release note (bug fix): Previously, CockroachDB would error when
receiving Geometry/Geography using binary parameters. This is now
resolved.

otan commented:
bors r=rafiss
